contracts/contracts/TACoApplication.sol

```diff
@@ -655,10 +657,18 @@ contract TACoApplication is
     }
 
     /**
-     * @notice Get all tokens delegated to the staking provider
+     * @notice Get eligible stake delegated to the staking provider
      */
-    function getEligibleAmount(StakingProviderInfo storage _info) internal view returns (uint96) {
-        return _info.authorized - _info.deauthorizing;
+    function eligibleStake(
+        address _stakingProvider,
+        uint256 _endDate
+    ) public view returns (uint96) {
+        StakingProviderInfo storage info = stakingProviderInfo[_stakingProvider];
+        uint96 eligibleAmount = info.endDeauthorization == 0 || info.endDeauthorization >= _endDate
+            ? info.authorized
+            : info.authorized - info.deauthorizing;
```

Comment:
Can we make this code a bit clearer?
```suggestion
        uint96 eligibleAmount = info.authorized;
        if (0 < info.endDeauthorization && info.endDeauthorization < _endDate) {
            eligibleAmount -= info.deauthorizing;
        }
```
Additionally, I'd introduce the restriction that `require(_endDate > block.timestamp);`

Reply:
✔️ 
i'm not sure about `require` to be honest, it's a call and even called with wrong data it won't break logic